diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57fd278..c1ab890 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -893,7 +894,6 @@ if(MINGW OR MSYS)
             -I ${CMAKE_CURRENT_BINARY_DIR}
             -o ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj
             -i ${CMAKE_CURRENT_SOURCE_DIR}/win32/zlib${SUFFIX}1.rc)
-    set(ZLIB_DLL_SRCS ${CMAKE_CURRENT_BINARY_DIR}/zlib1rc.obj)
 endif()
 
 set(ZLIB_ALL_SRCS ${ZLIB_SRCS} ${ZLIB_ARCH_HDRS} ${ZLIB_ARCH_SRCS} ${ZLIB_DLL_SRCS}
