diff --git a/include/__config b/include/__config
index 3cf23694f878..bcb3086248d3 100644
--- a/include/__config
+++ b/include/__config
@@ -228,7 +228,7 @@
 #endif
 
 // Need to detect which libc we're using if we're on Linux.
-#if defined(__linux__)
+#if defined(__linux__) && !defined(_LIBCPP_GLIBC_PREREQ)
 #  include <features.h>
 #  if defined(__GLIBC_PREREQ)
 #    define _LIBCPP_GLIBC_PREREQ(a, b) __GLIBC_PREREQ(a, b)
