diff --git a/lib/Driver/ToolChains/CommonArgs.cpp b/lib/Driver/ToolChains/CommonArgs.cpp
index 83cab3ac00cb..6a284384ab8f 100644
--- a/lib/Driver/ToolChains/CommonArgs.cpp
+++ b/lib/Driver/ToolChains/CommonArgs.cpp
@@ -1442,23 +1442,7 @@ static void AddUnwindLibrary(const ToolChain &TC, const Driver &D,
     break;
   }
   case ToolChain::UNW_CompilerRT:
-    if (TC.getTriple().isOSAIX()) {
-      // AIX only has libunwind as a shared library. So do not pass
-      // anything in if -static is specified.
-      if (LGT != LibGccType::StaticLibGcc)
-        CmdArgs.push_back("-lunwind");
-    } else if (LGT == LibGccType::StaticLibGcc) {
-      CmdArgs.push_back("-l:libunwind.a");
-    } else if (TC.getTriple().isOSCygMing()) {
-      if (LGT == LibGccType::SharedLibGcc)
-        CmdArgs.push_back("-l:libunwind.dll.a");
-      else
-        // Let the linker choose between libunwind.dll.a and libunwind.a
-        // depending on what's available, and depending on the -static flag
-        CmdArgs.push_back("-lunwind");
-    } else {
-      CmdArgs.push_back("-l:libunwind.so");
-    }
+    CmdArgs.push_back("-lunwind");
     break;
   }
 
