diff --git a/bootstrap.sh b/bootstrap.sh
index d84805425..047d3d17d 100755
--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -223,7 +223,7 @@ rm -f config.log
 if test "x$BJAM" = x; then
   $ECHO "Building B2 engine.."
   pwd=`pwd`
-  (cd "$my_dir/tools/build/src/engine" && ./build.sh)
+  (cd "$my_dir/tools/build/src/engine" && CXX="$CXX_FOR_BUILD" ./build.sh)
   if [ $? -ne 0 ]; then
       echo
       echo "Failed to build B2 build engine"
